### DISCUSSION TITLE
1-Wire rescan

### DISCUSSION THREAD

**Discussion by DetlefR** [arendst/Tasmota#16955] [Ideas]
Hello,

I use 5 DS18B20 on a SONOF 4Channel relay. It is working, but sometimes the thermometers don't send new values. I can fix the problem by a restart. But I think this is not the best way.
My idea. Is it possible to implement a command to rescan the 1-wire bus in the hope it works again after that.

Thanks
Detlef

**Comment by sfromis** [comment]
You should probably debug why it stalls, maybe you need to adjust your pullup. And if you use parasite power, better avoid that.

One technique for resetting could be if you control the power via a gpio pin. I've not done calculations on whether this would be feasible with 5 sensors, as a gpio can deliver 10 mA, or maybe a bit more.

**Reply by barbudor** [reply]
If more current is needed, a P-channel MOSFET driven by a GPIO as Relay Inverted would help:
"LOAD" are the DS18x20 power pin

![image](https://user-images.githubusercontent.com/2996042/198897522-609c298d-88d0-4810-a450-5000e5c1ed56.png)

**Comment by DetlefR** [comment]
Hello,
I have tried the  proposal from @sfromis with a gpio as power supply. According to the datasheet, one DS18B20 needs only 1 mA. On Sonoff 4Ch there are only three free ports. gpio2 is already DS18x20. gpio3 is now Relay_i 5. If the relay off the gpio supplies the thermometers with power. Is it on, 0.0°C shown. The thermometers are working again if the relay is switched off.

Thanks for the idea

**Reply by barbudor** [reply]
Why Relay_i ? 
Use "Relay" to power directly or "Relay_i" when through a transistor

**Comment by DetlefR** [comment]
@barbudor 
"Why Relay_i ?"
I prefer "PowerOnState 0". So it means that relay state is off but the DS19B20 is supplied. That looks strange, but it simplifies the handling after a power failure
